src/test/scala/dev/kamu/cli/MetadataRepositorySpec.scala

```diff
@@ -53,4 +61,63 @@ class MetadataRepositorySpec extends FunSuite with Matchers with KamuTestBase {
       actual3 shouldEqual Seq(dsC.id, dsB.id, dsA.id)
     }
   }
+
+  test(raw"'kamu add' from HTTP") {
+    withEmptyRepo { kamu =>
+      val expected = DatasetFactory.newRootDataset()
+
+      // create a temporary directory with the dataset to host
+      val serverDir = getRandomDir()
```

Comment:
this directory will not be cleaned up if test fails, you could simply create files in the directory already set up by `withEmptyRepo` and let this method clean them up for you instead

Reply:
Not sure how to get that directory path. `withEmptyRepo` does not change working directory path, so a relative path points outside of the temporary dir.

Reply:
Oh,I see the comment above, which means `KamuTestAdapter.config` should be public. Got it.